docs/user-guide/suppressions.md

```diff
@@ -0,0 +1,80 @@
+# Bulk suppressions
+
+Enabling a rule as `"error"` can be difficult when an established code-base already contains many violations that cannot be auto-fixed.
+Stylelint lets you _suppress_ those legacy violations so the rule is enforced **only** for new code. You can then clear the backlog at your own pace.
+
+> [!IMPORTANT]
+> Only rules configured with severity `"error"` are suppressed.
+> Violations reported as `"warning"` are **not** suppressed.
+
+## Suppressions file
+
+The first time you run Stylelint with `--suppress-all` (or `--suppress-rule`), the tool creates a file called `stylelint-suppressions.json` in the current working directory.
+This JSON file records which rules are being ignored in which files, so that subsequent runs only report new problems.
```

Comment:
question

> so that subsequent runs only report _new_ problems

Did you match ESLint behaviour?

> For example, if a file originally had 2 suppressed no-undef errors but now has 5, ESLint will display all 5 no-undef errors.

I wish we would display only 3, and not the 2 that were suppressed.
If you did, this part will have to be emended.
i.e. it's unclear

Reply:
Yes, we match ESLint's current behavior.

> For example, if a file originally had 2 suppressed no-undef errors but now has 5, ESLint will display all 5 no-undef errors.

See: [Introducing bulk suppressions - ESLint - Pluggable JavaScript Linter](https://eslint.org/blog/2025/04/introducing-bulk-suppressions/#getting-started)


> I wish we would display only 3, and not the 2 that were suppressed.

Ideally yes, but that would be difficult with the current implementation.

Reply:
Can you add a sentence that explains that then?
i.e. `so that subsequent runs only report new problems` is misleading (it will also report previously suppressed ones)